app/main/models/models.py

```diff
@@ -58,6 +58,52 @@ def format(self):
         }
 
 
+class Cluster(BaseModel):
+    """
+    Represents a specific cluster, which is a grouping of nodes.
+    """
+    __tablename__ = 'clusters'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String, nullable=False)
+    #represents the cartesian coordinates of the center of the cluster
+    x_position = db.Column(db.Integer)
+    y_position = db.Column(db.Integer)
+    # The project that holds this source
+    project_id = db.Column(db.Integer, db.ForeignKey('projects.id'),
+                           nullable=False)
+    parent_cluster_id = db.Column(db.Integer, db.ForeignKey('clusters.id'),
+                                  nullable=True)
+    #References to outermost clusters within a given cluster
+    child_clusters = db.relationship('Cluster',
+                                     backref=db.backref('parent_cluster',
+                                                        remote_side=[id]))
+    #References to sources not in another subcluster within a cluster
+    child_sources = db.relationship('Source', backref='cluster',
+                                    cascade='all, delete-orphan',
+                                    lazy=True)
+
+    def __repr__(self):
+        return f'<Cluster {self.id}: {self.name}>'
+
+    def format_long(self):
+        return {
+            'id': self.id,
+            'name': self.name,
+            'x_position': self.x_position,
+            'y_position': self.y_position,
+            'project_id': self.project_id
+        }
+
+    def format_short(self):
+        return {
+            'id': self.id,
+            'name': self.name,
+            'x_position': self.x_position,
+            'y_position': self.y_position,
+            'project_id': self.project_id
+        }
```

Comment:
These two functions are the same. You probably only need one called `format()`, and you may wanna return other properties as well.